napari/_qt/qt_main_window.py

```diff
@@ -443,13 +444,22 @@ def _add_file_menu(self):
 
     def _open_preferences(self):
         """Edit preferences from the menubar."""
-        win = PreferencesDialog(parent=self._qt_window)
-        win.resized.connect(self._qt_window._update_preferences_dialog_size)
+        if self._qt_window._preferences_dialog is None:
+            win = PreferencesDialog(parent=self._qt_window)
+            win.resized.connect(
+                self._qt_window._update_preferences_dialog_size
+            )
+
+            if self._qt_window._preferences_dialog_size:
+                win.resize(self._qt_window._preferences_dialog_size)
 
-        if self._qt_window._preferences_dialog_size:
-            win.resize(self._qt_window._preferences_dialog_size)
+            self._qt_window._preferences_dialog = win
+            win.closed.connect(self._on_preferences_closed)
+            win.show()
```

Comment:
Don't know if it's possible, but can we make the existing one be on top/ focus if it already exists ?

Reply:
>Don't know if it's possible, but can we make the existing one be on top/ focus if it already exists ?

Yes, that is what usually happens in other software that do not have modal preference.

Good catch.